src/main/java/bio/terra/stairway/HookWrapper.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.stairway;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HookWrapper {
+  private static final Logger logger = LoggerFactory.getLogger(HookWrapper.class);
+  private Stairway stairway;
+
+  public HookWrapper(Stairway stairway) {
+    this.stairway = stairway;
+  }
+
+  public HookAction startFlight(FlightContext context) {
+    return handleHook(context, HookType.startFlight);
+  }
+
+  public HookAction endFlight(FlightContext context) {
+    return handleHook(context, HookType.endFlight);
+  }
+
+  public HookAction startStep(FlightContext context) {
+    return handleHook(context, HookType.startStep);
+  }
+
+  public HookAction endStep(FlightContext context) {
+    return handleHook(context, HookType.endStep);
+  }
+
+  private HookAction handleHook(FlightContext context, HookType type) {
```

Comment:
Could we pass the method directly as a parameter instead of a `HookType` and then call it? Then we wouldn't need to do a switch or have a `HookType` object.

Reply:
I opted to just go the simpler route and have a bit of repetitive code, but thanks - good idea.